test/test/cpu/CpuTests.java

```diff
@@ -32,4 +43,52 @@ public void itimerTotal(TestProcess p) throws Exception {
         Output out = p.profile("-d 2 -e itimer -i 100ms --total -o collapsed");
         assertCloseTo(out.total(), 2_000_000_000, "itimer total should match profiling duration");
     }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX)
+    public void perfEventsWrongTargetCpu(TestProcess p) throws Exception {
+        pinCpu(p, 1);
+
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 2");
+        Assert.isEqual(out.total(), 0, "perf_events total should be 0 when the wrong CPU is targeted");
+    }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX)
+    public void perfEventsRightTargetCpu(TestProcess p) throws Exception {
+        pinCpu(p, 1);
+
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 1");
+        assertCloseTo(out.total(), 2_000_000_000, "perf_events total should match profiling duration");
+    }
+
+    @Test(mainClass = CpuBurner.class, os = Os.LINUX)
+    public void perfEventsWrongTargetCpuWithFdTransfer(TestProcess p) throws Exception {
+        pinCpu(p, 1);
+
+        Output out = p.profile("-d 2 -e cpu -i 100ms --total -o collapsed --target-cpu 2 --fdtransfer");
```

Comment:
I wonder how this will work on a host with 2 CPUs only.
Maybe it's better to use CPUs 0,1 instead of 1,2?

Reply:
Good point, I reduced all CPU indexes by 1: 07cb40f